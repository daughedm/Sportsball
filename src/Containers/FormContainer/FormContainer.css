.form-container {
  margin: auto;
  width: 75%;
}

.sportsball-logo {
  width: 650px;
  margin:140px auto 80px auto;
  display: block;

}

.loading {
  margin: auto;
  display: block;
}

.animated {
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

@-webkit-keyframes zoomIn {
  from {
    opacity: 0;
    -webkit-transform: scale3d(0.3, 0.3, 0.3);
    transform: scale3d(0.3, 0.3, 0.3);
  }
  50% {
    opacity: 1;
  }
}

@keyframes zoomIn {
  from {
    opacity: 0;
    -webkit-transform: scale3d(0.3, 0.3, 0.3);
    transform: scale3d(0.3, 0.3, 0.3);
  }
  50% {
    opacity: 1;
  }
}

.zoomIn {
  -webkit-animation-name: zoomIn;
  animation-name: zoomIn;
}