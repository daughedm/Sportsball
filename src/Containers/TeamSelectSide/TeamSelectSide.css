.unordered-list {
  list-style-type: none;
}

.scrollable-menu {
  height: auto;
  max-height: 170px;
  overflow-x: hidden;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #3C3C3A;
  margin-top: 25px;
  background-color: #F4F1E7;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  padding: 0 20px 0 20px;
}

.side-form {
  width: 100%;
  background-color: rgba(60, 60, 58, 0.18);
  border: 1px solid #F4F1E7;
  border-radius: 4px;
  margin: 60px auto;
  padding: 35px;
}

.form-header {
  color: #F26F44;
  font-family: "Noto Sans";
  font-size: 25px;
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 20px;
  letter-spacing: 1px;
}

.drop-down-button {
  display: block;
  width: 100%;
  height: 50px;
  border-radius: 4px;
  border: 1px solid #3C3C3A;
  margin-top: 25px;
  background-color: #F4F1E7;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  padding: 10px;
  font-size: 14px;
  color: #3C3C3A;
}

.submit-button {
  display: block;
  margin-top: 25px;
  width: 100%;
  height: 50px;
  border: none;
  border-radius: 4px;
  background: linear-gradient(0deg, #F26F44 11%, #F98059 52%, #FF8F6B 100%);
  color: #F4F1E7;
  font-size: 16px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  cursor: pointer;
}

.anchor-toggle {
  font-size: 14px;
  display: block;
  text-decoration: underline;
  color: #F4F1E7;
  font-style: italic;
  padding-top: 12px;
  cursor: pointer;
}

li a {
  text-decoration: none;
  font-size: 16px;
  color: #3C3C3A;
 }

.list-item {
 padding: 15px 0 15px 0;
 height: 50px;
 border-bottom: 1px solid #808080;
}

.unordered-list :hover {
  background-color: #ffe4db;
}

::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}

::-webkit-scrollbar-button {
  width: 0px;
  height: 0px;
}

::-webkit-scrollbar-thumb {
  background: #f26f44;
  border: 2px solid transparent;
  border-radius: 44px;
}

::-webkit-scrollbar-thumb:hover {
  background: #f26f44;
}

::-webkit-scrollbar-thumb:active {
  background: #f26f44;
}

::-webkit-scrollbar-track {
  background: #f4f1e7;
  border: 45px none #ffffff;
  border-radius: 44px;
}

::-webkit-scrollbar-track:hover {
  background: transparent;
}

::-webkit-scrollbar-track:active {
  background: transparent;
}

::-webkit-scrollbar-corner {
  background: transparent;
}