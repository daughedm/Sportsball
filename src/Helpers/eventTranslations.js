export const eventCleaner = (event) => {
  switch (event) {
    case 'aBK':
      return 'balk'
    case 'aCI':
      return 'catcher interference'
    case 'aD':
    case 'aDAD3':
    case 'aDAD4':
    case 'oDT3':
    case 'oDT4':
      return 'double'
    case 'aFCAD2':
    case 'aFCAD3':
    case 'aFCAD4':
    case 'oFC':
    case 'oFCT2':
    case 'oFCT3':
    case 'oFCT4':
      return 'fielders choice'
    case 'aHBP':
      return 'hit by pitch'
    case 'aHR':
      return 'homerun'
    case 'aIBB':
      return 'intentional walk'
    case 'aKLAD1':
    case 'aKLAD2':
    case 'aKLAD3':
    case 'aKLAD4':
      return 'strike looking'
    case 'aKSAD1':
    case 'aKSAD2':
    case 'aKSAD3':
    case 'aKSAD4':
      return 'strike swinging'
    case 'aROE':
    case 'aROEAD2':
    case 'aROEAD3':
    case 'aROEAD4':
      return 'error'
    case 'aS':
    case 'aSAD2':
    case 'aSAD3':
    case 'aSAD4':
      return 'single'
    case 'aSBAD1':
    case 'aSBAD2':
    case 'aSBAD3':
    case 'aSBAD4':
      return 'sacrifice bunt'
    case 'aSFAD1':
    case 'aSFAD2':
    case 'aSFAD3':
    case 'aSFAD4':
      return 'sacrifice fly'
    case 'aT':
    case 'aTAD4':
      return 'triple'
    case 'bAB':
    case 'bD':
    case 'bDB':
    case 'bIB':
    case 'bPO':
      return 'ball'
    case 'kF':
    case 'kFT':
      return 'foul ball'
    case 'kKL':
      return 'strike looking'
    case 'kKS':
      return 'strike swinging'
    case 'oBI':
      return 'hitter interference'
    case 'oFO':
      return 'fly out'
    case 'oGO':
      return 'ground out'
    case 'oKLT1':
    case 'oKLT2':
    case 'oKLT3':
    case 'oKLT4':
      return 'strike looking'
    case 'oKST1':
    case 'oKST2':
    case 'oKST3':
    case 'oKST4':
      return 'strike swinging'
    case 'oLO':
      return 'line out'
    case 'oOBB':
      return 'out of batters box'
    case 'oOp':
      return 'out on appeal'
    case 'oPO':
      return 'pop out'
    case 'oROET2':
    case 'oROET3':
    case 'oROET4':
      return 'reached on error'
    case 'oSB':
    case 'oSBT2':
    case 'oSBT3':
    case 'oSBT4':
      return 'sacrifice bunt'
    case 'oSF':
    case 'oSFT2':
    case 'oSFT3':
    case 'oSFT4':
      return 'sacrifice fly'
    case 'oST2':
    case 'oST3':
    case 'oST4':
      return 'single'
    case 'oTT4':
      return 'triple'
    default:
      return 'that play';
  }
}

export const inningCleaner = (inning) => {
  switch (inning) {
    case 1:
      return 'first';
    case 2:
      return 'second';
    case 3: 
      return 'third';
    case 4: 
      return 'forth';
    case 5:
      return 'fifth';
    case 6:
      return 'sixth';
    case 7:
      return 'seventh';
    case 8:
      return 'eighth';
    case 9:
      return 'ninth';
    case 10:
      return 'tenth';
    case 11:
      return 'eleventh';
    case 12:
      return 'twelth';
    case 13:
      return 'thirteenth';
    case 14:
      return 'fourteenth';
    case 15:
      return 'fifteenth';
    case 16:
      return 'sixteenth';
    case 17:
      return 'seventeenth';
    case 18:
      return 'eighteenth';
    case 19:
      return 'nineteenth';
    case 20:
      return 'twentieth';
    default:
      return 'inning'
  }
}

export const eventToSlang = (event) => {
  let randomHRNumber = Math.floor(Math.random() * slangHomerun.length);
  let randomSingleNumber = Math.floor(Math.random() * slangSingle.length);
  let randomDoubleNumber = Math.floor(Math.random() * slangDouble.length);
  let randomTripleNumber = Math.floor(Math.random() * slangTriple.length);
  
  switch (event) {
    case 'homerun':
      return slangHomerun[randomHRNumber]
    case 'single':
      return slangSingle[randomSingleNumber]
    case 'double':
      return slangDouble[randomDoubleNumber] 
    case 'triple':
      return slangTriple[randomTripleNumber]
  default:
      return event;
  }
}

let slangHomerun = ['dinger', 'moon shot', 'monster dong', 'donger', 'four bagger', 'jack', 'quadrangler', 'goner', 'tater', 'jonron', 'homerun', 'homer', 'mash', 'no-doubter' ];

let slangSingle = ['base knock', 'sencillo', 'bleeder', 'single', 'lace', 'rip' ]

let slangDouble = ['gapper', 'double', 'lace', 'rip', 'frozen rope', 'two bagger' ]

let slangTriple = ['three bagger', 'triple']