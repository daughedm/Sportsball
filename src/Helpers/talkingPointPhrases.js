const winningMessages = (yourTeam, score) => {
  return [`Wow, that game was sick! I can’t believe the ${yourTeam} won ${score}!`, `The ${yourTeam} just know how to win. `, `BAM, that game was sick! I can’t believe the ${yourTeam}  won ${score}!`, `SWEET, that game was sick! I can’t believe the ${yourTeam} won ${score}!`, `Chill, that game was sick! I can’t believe the ${yourTeam} won ${score}!`, `Damn, that game was sick! I can’t believe the ${yourTeam} won ${score}!`, 'The win today was a total team effort.', `The ${yourTeam} really played within themselves today.`, `The ${yourTeam} were due for a win.`, `Nothing comes easy in this league, great win for the ${yourTeam}.`, `The ${yourTeam} were clearly the more motivated team this game.`, `That was a statement win for the ${yourTeam}!`,`The ${score} margin of victory really shows the ${yourTeam} wanted it more.`];
};

const losingMessages = (yourTeam, score) => {
  return [`Damn, what a bummer! I can’t believe the ${yourTeam} lost ${score}!`, `Dude, how could the ${yourTeam} lose! They are better than that.`, `I thought the ${yourTeam} were going to pull it out, but they lost ${score}!`, `I'm not concerned, the ${yourTeam} have a lot of games to play`, `Damn, that game was sick! Too bad the ${yourTeam} lost ${score}!`, `The ${yourTeam} didn't have their best stuff today. They lost ${score}`, `On paper the ${yourTeam} were the team to beat but they just couldn't get that win.`, `The other team just wanted it more. We lost ${score}.`, `The ${yourTeam} need to rub some dirt on this loss.`, 'You can\'t teach a team to win, but they need to learn soon.', `The ${yourTeam} are still trying to figure out how to win.`, `The ${yourTeam} loss sure greased my glove!`];
};

const winningCloseMessages = (yourTeam, score) => {
  return [`Damn, that was such a close game I can’t believe the ${yourTeam} pulled it out ${score}!`];
};

const losingCloseMessages = (yourTeam, score) => {
  return [`Damn, losing ${score} is hard to swallow. I can’t believe the ${yourTeam} could'nt come out with the W.`];
};

const blowoutWinMessages = (yourTeam, score) => {
  return [`The ${yourTeam} dominated that game ${score}!`];
};

const blowoutLossMessages = (yourTeam, score) => {
  return [`The ${yourTeam} got blown out, ${score}!`];
};

const eventMessages = (yourTeam, event, inning, numberScored) => {
  return [`Did you see that ${event} in the ${inning} that scored ${numberScored}?`, `That ${event} in the ${inning} was so clutch.`, `I was so pumped about that ${event} in the ${inning}.`, `The ${inning} was such an explosive inning for us.`, `The ${event} in the ${inning} was amazing.`, `The ${event} in the ${inning} was very underrated.`];
};

const batterMessages = (first, last, stat) => {
  return [`${first} ${last} is having great year hitting ${stat}.`, `Can you believe ${first} ${last} is hitting ${stat}`, `${first} ${last} is having a career year hitting ${stat}.`, `${first} ${last}'s ${stat} average is best on the team.`, `I would never have guessed ${first} ${last} would be averaging ${stat}.`, `I hope ${first} ${last} can keep his ${stat} pace for the rest of the year.`];
};

const batterHRMessages = (first, last, stat) => {
  return [`${first} ${last} is having great year with ${stat} home runs.`];
};

const pitcherMessages = (first, last, stat) => {
  return [`${first} ${last} is so good, he has ${stat} wins this year already.`, `${first} ${last} is having a great year with ${stat} wins.`, `${first} ${last} is the real deal with his ${stat} wins this year.`, `Do you think ${first} ${last} can win much more than his ${stat} wins?`];
};

const pitcherERAMessages = (first, last, stat) => {
  return [`${first} ${last} is having great year with a ${stat} ERA.`];
};

export { winningMessages, losingMessages, eventMessages, batterMessages, pitcherMessages, winningCloseMessages, losingCloseMessages, blowoutLossMessages, blowoutWinMessages, batterHRMessages, pitcherERAMessages };


