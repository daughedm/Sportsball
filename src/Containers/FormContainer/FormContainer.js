import React, {Component} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import sportsballLogo from '../../Assets/images/logo_creme_home-02.svg';
import loading from '../../Assets/images/loading.gif';
import './FormContainer.css';
import TeamSelect from '../TeamSelect/TeamSelect';

export class FormContainer extends Component {
  
  render() {

    return ( 
      <div className="form-container">
        <img src={ sportsballLogo } className="sportsball-logo animated zoomIn" alt="Sportsball logo"/>
        {
          !this.props.gameSummaries.length && !this.props.boxScores.length ? (
            <img className="loading" src={ loading } alt="loading"/>
          ) : (
            < TeamSelect />
          )}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  gameSummaries: state.gameSummaries,
  boxScores: state.boxScores
});

FormContainer.propTypes = {
  gameSummaries: PropTypes.object,
  boxScores: PropTypes.object

};

export default connect(mapStateToProps, null)(FormContainer);




