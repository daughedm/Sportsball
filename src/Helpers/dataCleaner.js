import * as prases from './talkingPointPhrases';

const yesterdaysDate = () => {
  let todayTimeStamp = new Date();
  let oneDayTimeStamp = 1000 * 60 * 60 * 24;
  let diff = todayTimeStamp - oneDayTimeStamp;
  let yesterdayDate = new Date(diff);
  let year = yesterdayDate.getFullYear()
  let month = yesterdayDate.getMonth() + 1;
  let day = yesterdayDate.getDate();

  if (month < 10) {
    month = '0' + month;
  }
  if (day < 10) {
    day = '0' + day;
  }

  let yesterdayString = year + '/' + month + '/' + day;
  return yesterdayString;
}

const gameSummaryCleaner = (rawSummaryData) => {
  const cleanedSummary = rawSummaryData.league.games.map(ballGame => {
    return {
      homeTeam: ballGame.game.home.name,
      awayTeam: ballGame.game.away.name,
      homeTeamRuns: ballGame.game.home.runs,
      awayTeamRuns: ballGame.game.away.runs,
      winningTeam: ballGame.game.home.runs > ballGame.game.away.runs ? ballGame.game.home.name : ballGame.game.away.name,
      losingTeam: ballGame.game.home.runs < ballGame.game.away.runs ? ballGame.game.home.name : ballGame.game.away.name
    }
  })
  return cleanedSummary;
}

const boxScoresCleaner = (rawBoxScoresData) => {
  const cleanedBoxScores = rawBoxScoresData.league.games.map(ballgame => {
    return {
      homeTeam: ballgame.game.home.name,
      awayTeam: ballgame.game.away.name,
      homeTeamEvents: ballgame.game.home.events,
      awayTeamEvents: ballgame.game.home.events
    }
  })
  return cleanedBoxScores;
} 

const findGameSummary = (yourTeam) => {
  let singleGame = cleanData.find(ballGame => {
    return ballGame.homeTeam.includes(yourTeam) || ballGame.awayTeam.includes(yourTeam)
  })
  if (singleGame === undefined) {
    singleGame = `The ${yourTeam} didn't play!`
  }
  return singleGame;
}

const findGameBoxScore = (yourTeam) => {
  let singleGame = cleanData.find(ballGame => {
    return ballGame.homeTeam.includes(yourTeam) || ballGame.awayTeam.includes(yourTeam)
  })
  if (singleGame === undefined) {
    singleGame = `The ${yourTeam} didn't play!`
  }
  return singleGame;
}

// const findGame = (yourTeam) => {
//   const singleGame = gameSummaryCleaner.find(ballGame => {
//    return ballGame.homeTeam.includes(yourTeam) || ballGame.awayTeam.includes(yourTeam)
//   })
//   return singleGame;
// }

// const firstTalkingPoint = (yourTeam) => {
//   const teamsGame = findGame(yourTeam);
//   let score = `${teamsGame.homeTeamRuns} - ${teamsGame.awayTeamRuns}`;
//   let randNum;
//   let message;

//   if (teamsGame.winningTeam === yourTeam) {
//     randNum = Math.floor(Math.random() * winningMessages(yourTeam, score).length);
//     console.log(randNum)
//     message = winningMessages(yourTeam, score)[randNum];
//   } else if (teamsGame.winningTeam !== yourTeam && teamsGame.winningTeam !== undefined) {
//     randNum = Math.floor(Math.random() * losingMessages(yourTeam, score).length);
//     message = losingMessages(yourTeam, score)[randNum];
//   } else {
//     message = `The ${yourTeam} didn't play today!`;
//   }
//   return message;
// }

export { 
  yesterdaysDate,
  gameSummaryCleaner,
  boxScoresCleaner
};