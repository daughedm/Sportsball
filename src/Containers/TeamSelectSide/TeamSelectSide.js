import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { findGame, teamStatCleaner } from '../../Helpers/dataCleaner'
import { addSelectedTeamToStore } from '../../Actions/userActions';
import './TeamSelectSide.css';
import { addSingleSummaryToStore, addSingleBoxScoreToStore, addTeamStatsToStore } from '../../Actions';
import { teamStatsFetch } from '../../ApiCall/ApiCall';
import TeamScroll from './../TeamScroll/TeamScroll';

export class TeamSelectSide extends Component {

  render() {
    return ( 
      <form
        className = "side-form"
        type = "submit"
      >
        <h3 className = "form-header" > Switch Teams </h3> 
          <TeamScroll />
      </form>
    )
  }
}

TeamSelectSide.propTypes = {

}

export default TeamSelectSide
