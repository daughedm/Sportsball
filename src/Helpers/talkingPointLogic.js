import { winningMessages, losingMessages, eventMessages, batterMessages, pitcherMessages } from './talkingPointPhrases';
import { eventCleaner, inningCleaner, eventToSlang } from './eventTranslations';
import { filterBatters, filterPitchers, findHighestAvg, findMostWins } from '../Helpers/dataCleaner';

const firstTalkingPoint = (yourTeam, mlbData) => {
  let score = `${mlbData.homeTeamRuns} - ${mlbData.awayTeamRuns}`;
  let randNum;
  let message;

  if (mlbData.winningTeam === yourTeam) {
    randNum = Math.floor(Math.random() * winningMessages(yourTeam, score).length);
    message = winningMessages(yourTeam, score)[randNum];
  } else if (mlbData.winningTeam !== yourTeam && mlbData.winningTeam !== undefined) {
    randNum = Math.floor(Math.random() * losingMessages(yourTeam, score).length);
    message = losingMessages(yourTeam, score)[randNum];
  } else {
    message = `The ${yourTeam} didn't play today!`;
  }
  return message;
}

const secondTalkingPoint = (yourTeam, mlbData) => {
  let message;
  let randNumMessage = Math.floor(Math.random() * eventMessages('x', 'x', 'x', 'x').length);
  let randomNumEvent;

  if (mlbData.homeTeam === yourTeam) {
    randomNumEvent = Math.floor(Math.random() * mlbData.homeTeamEvents.length)
    //figure out what happens when there are no events.
    let event = eventToSlang(eventCleaner(mlbData.homeTeamEvents[randomNumEvent].hitter_outcome));
    let inning = inningCleaner(mlbData.homeTeamEvents[randomNumEvent].inning);
    let runsScored = mlbData.homeTeamEvents[randomNumEvent].runners.length;
    message = eventMessages(yourTeam, event, inning, runsScored)[randNumMessage];
    if (message === undefined) {
      message = 'That was such a boring game. Literally nothing eventful happened'
    }

  } else if (mlbData.awayTeam === yourTeam) {
    randomNumEvent = Math.floor(Math.random() * mlbData.awayTeamEvents.length)
    let event = eventToSlang(eventCleaner(mlbData.awayTeamEvents[randomNumEvent].hitter_outcome));
    let inning = inningCleaner(mlbData.awayTeamEvents[randomNumEvent].inning);
    let runsScored = mlbData.awayTeamEvents[randomNumEvent].runners.length;
    message = eventMessages(yourTeam, event, inning, runsScored)[randNumMessage];
    if (message === undefined) {
      message = 'That was such a boring game. Literally nothing eventful happened'
    }
  } else {
    message = `Talk about the weather or something, I dunno.`;
  }
  return message;
}

const thirdTalkingPoint = (yourTeam, teamStats) => {
  let message;
  let randomBatterNumber;
  let randomPitcherNumber;
  let randNum = Math.floor(Math.random() * 2)
  let bestBatter = findHighestAvg(filterBatters(teamStats, yourTeam));
  let bestPitcher = findMostWins(filterPitchers(teamStats, yourTeam));

  if(randNum < 1) {
    randomBatterNumber = Math.floor(Math.random() * batterMessages('x', 'x', 'x').length);
    message = batterMessages(bestBatter.first_name, bestBatter.last_name, bestBatter.statistics.hitting.overall.avg)[randomBatterNumber]
  } 
  else if(randNum >= 1) {
    randomPitcherNumber = Math.floor(Math.random() * pitcherMessages('x', 'x', 'x').length);
    message = pitcherMessages(bestPitcher.first_name, bestPitcher.last_name, bestPitcher.statistics.pitching.overall.games.win)[randomPitcherNumber]
  }
return message
}

export { firstTalkingPoint, secondTalkingPoint, thirdTalkingPoint };