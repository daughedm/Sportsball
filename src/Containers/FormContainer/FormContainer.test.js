import { FormContainer, mapStateToProps } from "./FormContainer";
import React from "react";
import { shallow } from "enzyme";
import { connect } from 'react-redux';
import { yesterdaysDate } from '../../Helpers/dataCleaner';
import { winningMessages, losingMessages } from '../../Helpers/talkingPointPhrases';
import { firstTalkingPoint, secondTalkingPoint, thirdTalkingPoint } from '../../Helpers/talkingPointLogic';
import TeamSelectSide from '../TeamSelectSide/TeamSelectSide';
import {
  mockCleanSummary,
  mockCleanBoxScores,
  mockCleanTeamStats
} from './../../mockData/mockCleanData';

const mockMath = Object.create(global.Math);
mockMath.random = () => 0;
global.Math = mockMath;

describe('FormContainer',() => {
  let wrapper;
  let mockSingleGameSummary;
  let mockSingleBoxScore;
  
  beforeEach(() => {
    mockSingleGameSummary = mockCleanSummary;
    mockSingleBoxScore = mockCleanBoxScores;
    
    wrapper = shallow(<FormContainer 
      gameSummaries={mockSingleGameSummary}
      boxScores={mockSingleBoxScore}
    /> );
  })


  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
})