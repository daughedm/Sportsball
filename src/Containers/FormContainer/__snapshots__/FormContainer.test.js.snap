// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormContainer
    boxScores={
        Object {
            "awayTeam": "Cubs",
            "awayTeamEvents": Array [
              Object {},
              Object {},
            ],
            "homeTeam": "Indians",
            "homeTeamEvents": Array [
              Object {
                "runners": Array [
                  1,
                  2,
                  3,
                ],
              },
              Object {
                "runners": Array [
                  1,
                  2,
                  3,
                ],
              },
            ],
          }
    }
    gameSummaries={
        Object {
            "awayTeam": "Cubs",
            "awayTeamRuns": 0,
            "homeTeam": "Indians",
            "homeTeamRuns": 100,
            "losingTeam": "Cubs",
            "winningTeam": "Indians",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Sportsball logo"
          className="sportsball-logo animated zoomIn"
          src="logo_creme_home-02.svg"
/>,
        <img
          alt="loading"
          className="loading"
          src="loading.gif"
/>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Sportsball logo",
          "className": "sportsball-logo animated zoomIn",
          "src": "logo_creme_home-02.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading",
          "className": "loading",
          "src": "loading.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Sportsball logo"
            className="sportsball-logo animated zoomIn"
            src="logo_creme_home-02.svg"
/>,
          <img
            alt="loading"
            className="loading"
            src="loading.gif"
/>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Sportsball logo",
            "className": "sportsball-logo animated zoomIn",
            "src": "logo_creme_home-02.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "className": "loading",
            "src": "loading.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
